api/c/indigo/src/indigo_misc.cpp

```diff
@@ -1419,118 +1417,32 @@ CEXPORT const char* indigoCheckObj2(int item, const char* check_flags)
-            ch.checkBaseReaction(brxn);
-        }
-        else if (IndigoAtom::is(obj))
-        {
-            StructureChecker ch(out);
-            ch.parseCheckTypes(props);
-
-            IndigoAtom& ia = IndigoAtom::cast(obj);
-            QS_DEF(Array<int>, atoms);
-            atoms.clear();
-            atoms.push(ia.getIndex() + 1);
-            ch.addAtomSelection(atoms);
-
-            if (ia.mol.isQueryMolecule())
-            {
-                QueryMolecule& qmol = ia.mol.asQueryMolecule();
-                ch.checkQueryMolecule(qmol);
-            }
-            else
-            {
-                Molecule& mol = ia.mol.asMolecule();
-                ch.checkMolecule(mol);
-            }
-        }
-        else if (IndigoBond::is(obj))
-        {
-            StructureChecker ch(out);
-            ch.parseCheckTypes(props);
-
-            IndigoBond& ib = IndigoBond::cast(obj);
-            QS_DEF(Array<int>, bonds);
-            bonds.clear();
-            bonds.push(ib.getIndex() + 1);
-            ch.addBondSelection(bonds);
-
-            if (ib.mol.isQueryMolecule())
-            {
-                QueryMolecule& qmol = ib.mol.asQueryMolecule();
-                ch.checkQueryMolecule(qmol);
-            }
-            else
-            {
-                Molecule& mol = ib.mol.asMolecule();
-                ch.checkMolecule(mol);
-            }
-        }
-        out.writeChar(0);
-        return tmp.string.ptr();
-    }
-    INDIGO_END(0);
-}
-
-CEXPORT const char* indigoCheckStructure(const char* structure, const char* props)
-{
-    INDIGO_BEGIN
-    {
-        auto& tmp = self.getThreadTmpData();
-        ArrayOutput out(tmp.string);
-
-        try
-        {
-            int item = indigoLoadStructureFromString(structure, "");
-            if (item > 0)
-                out.writeString(indigoCheck(item, props));
-            else
-                out.writeString("{\"LOAD\":{\"message\":\"Error at loading structure\"}}");
-        }
-        catch (Exception& e)
-        {
-            out.printf("{\"LOAD\":{\"message\":\"Error at loading structure. %s\"}}", e.message());
-        }
-
-        out.writeChar(0);
-        return tmp.string.ptr();
-    }
-    INDIGO_END(0);
+CEXPORT const char* indigoCheckStructure(const char* structure, const char* props)
+{
+    INDIGO_BEGIN
+    {
+        auto& tmp = self.getThreadTmpData();
+        ArrayOutput out(tmp.string);
+
+        try
+        {
+            int item = indigoLoadStructureFromString(structure, "");
+            if (item > 0)
+                out.writeString(indigoCheckObj(item, props));
+            else
+                out.writeString("{\"LOAD\":{\"message\":\"Error at loading structure\"}}");
+        }
+        catch (Exception& e)
+        {
+            out.printf("{\"LOAD\":{\"message\":\"Error at loading structure. %s\"}}", e.message());
+        }
+
+        out.writeChar(0);
+        return tmp.string.ptr();
+    }
+    INDIGO_END(0);

```

Comment:
Better use `nullptr` here.

Reply:
This is not nullptr. Here we return pointer to the null-terminated string. 0 here - means 0x0 character.

Reply:
@even1024, you are mistaken. 0 is the value returned by the function. Function return type is `const char *`. This means it returns null-pointer. 
If you want to return empty string, type: `INDIGO_END("");`